### PR TITLE
Subaru: frequency based messaging

### PR DESCRIPTION
**Pull request by jnewb1**
**Description**
Changes Subaru messaging to use frequency rather than relying on the counter variable from eyesight. This paradigm change will be required for gen2 long, where we disable or unplug the eyesight camera.

**Verification**
Tested on my Subaru Legacy 22

**Route**
d9df6f87e8feff94|2023-04-29--10-08-40--0

### CODE REVIEW

File: selfdrive/car/subaru/carcontroller.py [hunk 1]
```diff
@@ -10,16 +10,23 @@ def __init__(self, dbc_name, CP, VM):
...
-    self.es_distance_cnt = -1
-    self.es_dashstatus_cnt = -1
-    self.infotainmentstatus_cnt = -1
     self.cruise_button_prev = 0
     self.last_cancel_frame = 0
 
     self.p = CarControllerParams(CP)
     self.packer = CANPacker(DBC[CP.carFingerprint]['pt'])
 
+  def should_send_message(self, message_name):
```
Comment (sshane) [inline selfdrive/car/subaru/carcontroller.py:19]: let's do what other car ports do and just check the freq in the if statement, we can do global clean ups like this later

Comment (jnewb1) [reply selfdrive/car/subaru/carcontroller.py]: done

**Comment by jnewb1** [comment]
we should probably merge https://github.com/commaai/openpilot/pull/25345 first, then ill add all the long frequency stuff to this PR

**Comment by martinl** [comment]
Looks like all three 10Hz messages can be behind one `if self.frame % 10 == 0:`

**Comment by martinl** [comment]
Also, about original implementation with message frequencies array - message frequencies are already defined it carstate get_can_parser checks, is there an easy way to use them from carcontroller in future cleanup PR for all brands?

**Comment by martinl** [comment]
Also tested and working with global gen1: `8de015561e1ea4a0|2023-04-30--20-34-11`

**Comment by sshane** [comment]
@jnewb1 can you open a PR to make the counters use openpilot's instead of the forwarded messages'?

**Pull request merged**